javascript/src/router.js

```diff
@@ -3,25 +3,45 @@
  * See https://github.com/visionmedia/page.js
  */
 import page from 'page.js';
+import url from 'url';
 
 /**
- * Wrapper for `page.show()` with SilverStripe specific behaviour.
+ * Add leading slash to base-relative urls, as required by Page.js
+ *
+ * @param {string} path
+ * @return {string} Normalised path
  */
-function show(pageShow) {
-  return (path, state, dispatch, push) => {
-    // Normalise `path` so that pattern matching is more robust.
-    // For example if your route is '/pages' it should match when `path` is
-    // 'http://foo.com/admin/pages', '/pages', and 'pages'.
-    const el = document.createElement('a');
-    let pathWithSearch;
-    el.href = path;
-    pathWithSearch = el.pathname;
-    if (el.search) {
-      pathWithSearch += el.search;
+function normalise(path) {
```

Comment:
`normalise` feels a bit too generic a name, maybe something `pathRelativeToBase` or `baseRelativePath` would be better?